### DISCUSSION TITLE
Add a new field to input.graphql to identify if the function is running in Draft Order context

### DISCUSSION THREAD

**Discussion by AndreaMorone** [Shopify/function-examples#582] [Feedback: Delivery Customization API]
Would it be possible to add a new field in the input.graphql schema to indicate if a function is running on a draft order or a normal checkout? This would provide greater control for customizations.

Should be a 0-cost field, ensuring no impact on existing logic builders or workflows.

This would allow precise logic targeting draft orders vs. normal checkouts (we are receiving a lot of requests from merchants that do not want run customizations in the Shopify admin).

**Comment by HardikVansia** [comment]
I agree with @AndreaMorone

**Comment by thosakwe** [comment]
I recently received this request from a merchant:

```
Can a discount code be enabled for usse by Shopify admins only? Meaning, have the code not able to be entered at checkut, only added to draft orders by whomever works for the shopify store.
```

Being able to query whether a Shopify Function is running in a draft order context (without additional input query cost) would make this possible.
